### PR TITLE
Update lvgl 8.3.11

### PR DESCRIPTION
**Pull request by faxe1008**
Update LVGL module to version 8.3.11. Full release notes here:
[CHANGELOG.md](https://github.com/lvgl/lvgl/blob/release/v8.3/docs/CHANGELOG.md).

Substantial fixes:
- fix(disp): fix memory leak lv_scr_load_anim with auto_del and time=0 [1caafc5](https://github.com/lvgl/lvgl/commit/1caafc55dde46e1b7e3d17d8c5349fbf7cccba9f)
- fix(btnmatrix): fix array out of bounds addressing with groups and no buttons [edd5ad2](https://github.com/lvgl/lvgl/commit/edd5ad28ad74d1025db0b442cfa390d8c01ae006)
- fix(chart): fix lv_chart_get_point_pos_by_id [f9ffcc9](https://github.com/lvgl/lvgl/commit/f9ffcc9d8e11beb369dcbab0945ca85eab8f77b2)
- fix(chart): fix division by zero if there are no ticks [67b3011](https://github.com/lvgl/lvgl/commit/67b3011f835b08c77ff60ae2d1e6592a710ca2b1)

### CODE REVIEW

**Comment by faxe1008** [comment]
This commit introduced a regression in regards to background opacity behaviour when loading screens with FADE_ANIM for me:
```
commit ebd3537c2c92de1532a4b4465f4f08f8f5ff556b
Author: _VIFEXTech <vifextech@foxmail.com>
Date:   Mon Aug 14 13:57:19 2023 +0800

    feat(style): backport opa_layered
```

File: src/core/lv_obj_draw.c [hunk 1]
```diff
@@ -181,14 +180,12 @@ void lv_obj_init_draw_rect_dsc(lv_obj_t * obj, uint32_t part, lv_draw_rect_dsc_t
     }
 #endif
 
-    if(part != LV_PART_MAIN) {
```
Comment (faxe1008) [inline src/core/lv_obj_draw.c:184]: This check is still needed, to prevent opacity glitches.

Comment (faxe1008) [reply src/core/lv_obj_draw.c:184]: Created upstream issue:
[https://github.com/lvgl/lvgl/issues/4611](https://github.com/lvgl/lvgl/issues/4611)

Comment (faxe1008) [reply src/core/lv_obj_draw.c:184]: Associated PR:
[https://github.com/lvgl/lvgl/pull/4617](https://github.com/lvgl/lvgl/pull/4617)

**Comment by faxe1008** [comment]
Yesterday v8.3.11 was released. I added the respective new commits to this PR. So far I could not spot any regressions.

**Comment by pdgendt** [comment]
Do we still need a `zephyr` branch to merge to, or could we simply track upstream releases?

**Comment by fabiobaltieri** [comment]
> Do we still need a `zephyr` branch to merge to, or could we simply track upstream releases?

Well we need a branch to merge to, are you proposing to ditch the repo entirely? Pretty sure we don't what that even if it follows upstream 1:1, it's useful to have anyway in case we want to diverge temporarily and converge back down the line.

**Comment by fabiobaltieri** [comment]
> Not ditching, simply follow upstream 1:1, like we do for `openthread`.

Ok so you mean: drop the `zephyr` branch, instead use a `main` branch that tracks upstream? That'd be cool... it could still be called `zephyr` :-)

**Comment by pdgendt** [comment]
> Ok so you mean: drop the `zephyr` branch, instead use a `main` branch that tracks upstream? That'd be cool... it could still be called `zephyr` :-)

Right, but now we have to keep adding merge commits simply for the `zephyr` folder?

**Comment by fabiobaltieri** [comment]
Oh I see, yeah it'd make sense to use main, or maybe just tag it so git does not drop the refs and then force push it.

File: Kconfig
Comment (faxe1008) [inline Kconfig:1]: Kconfig guard introduced in [7c61a4cec26402d20c845c95dcad0e39dcd319f8](https://github.com/zephyrproject-rtos/lvgl/commit/7c61a4cec26402d20c845c95dcad0e39dcd319f8) is missing

Comment (faxe1008) [reply Kconfig:1]: Fixed

**Comment by faxe1008** [comment]
@pdgendt @fabiobaltieri Sure it would be doable to set up a branch which just tracks upstream. The differences between upstream and our fork mostly comes down to Kconfig changes. Btw. what I also found is that upstream LVGL also has zephyr `module.yaml` already :upside_down_face: [https://github.com/lvgl/lvgl/blob/master/env_support/zephyr/module.yml](https://github.com/lvgl/lvgl/blob/master/env_support/zephyr/module.yml)

Maybe we can sync our module file to upstream and have it be part of v9.0, so that we can eventually ditch the zephyr folder?

**Comment by fabiobaltieri** [comment]
> Maybe we can sync our module file to upstream and have it be part of v9.0, so that we can eventually ditch the zephyr folder?

Ideal.

**Comment by faxe1008** [comment]
Ping @fabiobaltieri  @pdgendt do you maybe have an idea if its possible that the modules `module.yml` lives inside the `env_support` folder and not within root of the repo?

**Comment by fabiobaltieri** [comment]
> Ping @fabiobaltieri @pdgendt do you maybe have an idea if its possible that the modules `module.yml` lives inside the `env_support` folder and not within root of the repo?

It looks like the location is hardcoded https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/zephyr_module.py#L179-L180 so with the current code base that is not possible, that said, it may be reasonable to expand the search path, maybe look for a `module.yml` file in any directory named `zephyr` in the module or something like that. Are you thinking about trying to go 100% in sync with upstream?

**Comment by faxe1008** [comment]
@fabiobaltieri Thanks for your confirmation. 100% is not feasable since there are Kconfig defaults that we have set differently. I am content to have the merging strat etc. as is. I think that @pdgendt raised the question of tracking 1:1 originally.
As I see it's not really possible to track upstream directly as we already diverged slightly and from what I see other modules also still perform merges into a dedicated zephyr branch.

Anyways if there is feasable solution I'd love to hear it. This should not be a blocker for this update imo.

**Comment by fabiobaltieri** [comment]
> Anyways if there is feasable solution I'd love to hear it. This should not be a blocker for this update imo.

If that's the only delta from upstream I think it may make sense to try and push to make the module file detection a bit smarter, but if we are going to have other patches on top of upstream anyway it's probably not worth the hassle IMO.

**Review by pdgendt — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
@faxe1008 is this good to go?

**Pull request merged**